### DISCUSSION TITLE
Phase 2 will only establish from one side

### DISCUSSION THREAD

**Discussion by doug-tucker-nav** [strongswan/strongswan#1393] [General]
I have a client we set up a vpn tunnel with.  They are using a cisco, I am using strongswan 5.9.4.  If clients from my side initiate a connection to them, the tunnel builds and traffic flows as expected.  However, if the tunnel is not currently up and they initiate a connection to us, the tunnels begins building, get phase 1  but phase 2 fails.  What I am seeing in the logs is an invalid payload length followed by unable to decrypt.  It tries several times and then gives up and removes the connection.  Lots of googling lead me to quite a few articles but none with solution.  I'm really not sure where to go on this one.  We do match on the cryto settings obviously as we establish when traffic initiates from my side.

Nov  2 15:44:41 ip-1-1-1-1 charon: 10[ENC] generating QUICK_MODE response 3972963768 [ HASH SA No KE ID ID ]
Nov  2 15:44:41 ip-1-1-1-1 charon: 10[NET] sending packet: from 1.1.1.1[4500] to 2.2.2.2[4500] (460 bytes)
Nov  2 15:44:42 ip-1-1-1-1 charon: 12[NET] received packet: from 2.2.2.2[4500] to 1.1.1.1[4500] (76 bytes)
Nov  2 15:44:42 ip-1-1-1-1 charon: 12[ENC] invalid HASH_V1 payload length, decryption failed?
Nov  2 15:44:42 ip-1-1-1-1 charon: 12[ENC] could not decrypt payloads

**Comment by Thermi** [comment]
Possible their PSK is just wrong or incorrectly configured somehow. Entirely depends on their configuration. Are you initiating with IKEv2 and they with IKEv1?

**Comment by doug-tucker-nav** [comment]
If the PSK or crypto didn't match, it should not work when I initiate it correct? Traffic from me results in an immediate and complete tunnel build.  This is ikev1 for phase1.  Phase1 does complete with the PSK from them to me as well.





________________________________

**Reply by Thermi** [reply]
It should, yes, but at times it happens that configurations are asymmetric for some reason.
It just happened at another setup with a Checkpoint VSX.

Can you provide full configuration and output of `ipsec statusall` and `swanctl -L`, maybe logs too, like in #196?

**Comment by doug-tucker-nav** [comment]
Thank you.  Here is what you have requested.

[root@ip-10-221-119-237 etc]# ipsec statusall
Status of IKE charon daemon (strongSwan 5.9.4, Linux 5.10.147-133.644.amzn2.x86_64, x86_64):
  uptime: 2 hours, since Nov 02 17:25:46 2022
  malloc: sbrk 2576384, mmap 0, used 694880, free 1881504
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 0
  loaded plugins: charon aesni aes des rc2 sha2 sha3 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl gcrypt fips-prf gmp curve25519 chapoly xcbc cmac hmac gcm drbg attr kernel-netlink resolve socket-default connmark stroke vici updown xauth-generic counters
Listening IP addresses:
  10.221.119.237
Connections:
img-10-10-0-0-16-10-221-112-0-22:  10.221.119.237...88.88.88.88  IKEv1, dpddelay=15s
img-10-10-0-0-16-10-221-112-0-22:   local:  [99.99.99.99] uses pre-shared key authentication
img-10-10-0-0-16-10-221-112-0-22:   remote: [88.88.88.88] uses pre-shared key authentication
img-10-10-0-0-16-10-221-112-0-22:   child:  10.221.112.0/22 === 10.10.0.0/16 TUNNEL, dpdaction=clear
img-10-1-75-0-24-10-221-112-0-22:   child:  10.221.112.0/22 === 10.1.75.0/24 TUNNEL, dpdaction=clear
img-10-46-0-0-16-10-221-112-0-22:   child:  10.221.112.0/22 === 10.46.0.0/16 TUNNEL, dpdaction=clear
img-10-20-10-0-24-10-221-112-0-22:   child:  10.221.112.0/22 === 10.20.10.0/24 TUNNEL, dpdaction=clear
Routed Connections:
img-10-20-10-0-24-10-221-112-0-22{4}:  ROUTED, TUNNEL, reqid 4
img-10-20-10-0-24-10-221-112-0-22{4}:   10.221.112.0/22 === 10.20.10.0/24
img-10-46-0-0-16-10-221-112-0-22{3}:  ROUTED, TUNNEL, reqid 3
img-10-46-0-0-16-10-221-112-0-22{3}:   10.221.112.0/22 === 10.46.0.0/16
img-10-1-75-0-24-10-221-112-0-22{2}:  ROUTED, TUNNEL, reqid 2
img-10-1-75-0-24-10-221-112-0-22{2}:   10.221.112.0/22 === 10.1.75.0/24
img-10-10-0-0-16-10-221-112-0-22{1}:  ROUTED, TUNNEL, reqid 1
img-10-10-0-0-16-10-221-112-0-22{1}:   10.221.112.0/22 === 10.10.0.0/16
Security Associations (0 up, 0 connecting):
  none

[root@ip-10-221-119-237 etc]# swanctl -L
img-10-10-0-0-16-10-221-112-0-22: IKEv1, reauthentication every 28260s, dpd delay 15s
  local:  10.221.119.237
  remote: 88.88.88.88
  local pre-shared key authentication:
    id: 99.99.99.99
  remote pre-shared key authentication:
    id: 88.88.88.88
  img-10-10-0-0-16-10-221-112-0-22: TUNNEL, rekeying every 3060s, dpd action is clear
    local:  10.221.112.0/22
    remote: 10.10.0.0/16
  img-10-1-75-0-24-10-221-112-0-22: TUNNEL, rekeying every 3060s, dpd action is clear
    local:  10.221.112.0/22
    remote: 10.1.75.0/24
  img-10-46-0-0-16-10-221-112-0-22: TUNNEL, rekeying every 3060s, dpd action is clear
    local:  10.221.112.0/22
    remote: 10.46.0.0/16
  img-10-20-10-0-24-10-221-112-0-22: TUNNEL, rekeying every 3060s, dpd action is clear
    local:  10.221.112.0/22
    remote: 10.20.10.0/24



[root@ip-10-221-119-237 etc]# cat ipsec.conf
# ipsec.conf - strongSwan IPsec configuration file
# basic configuration

config setup
        # strictcrlpolicy=yes
        # uniqueids = no
        # charondebug = "ike 2,chd 3, enc 2"

# Add connections here.

############################################################
##      Common configuration
############################################################

conn img-common
        left            = 10.221.119.237
        leftid          = 99.99.99.99
        leftsubnet      = 10.221.112.0/22
        leftfirewall    = no

        auto            = ignore
        authby          = secret
        closeaction     = clear
        fragmentation   = yes
        keyexchange     = ikev1
        reauth          = yes
        forceencaps     = yes
        rekey           = yes
        installpolicy   = yes
        type            = tunnel
        dpdaction       = clear
        dpddelay        = 15s
        dpdtimeout      = 60s
        ikelifetime     = 28800s
        lifetime        = 3600s
        ike             = aes256-sha256-modp2048
        esp             = aes256-sha256-modp2048
        aggressive      = no

conn img-10-10-0-0-16-10-221-112-0-22
        also            = img-common
        rightfirewall   = no
        auto            = route
        right           = 88.88.88.88
        rightsubnet     = 10.10.0.0/16

conn img-10-1-75-0-24-10-221-112-0-22
        also            = img-common
        rightfirewall   = no
        auto            = route
        right           = 88.88.88.88
        rightsubnet     = 10.1.75.0/24

conn img-10-46-0-0-16-10-221-112-0-22
        also            = img-common
        rightfirewall   = no
        auto            = route
        right           = 88.88.88.88
        rightsubnet     = 10.46.0.0/16

conn img-10-20-10-0-24-10-221-112-0-22
        also            = img-common
        rightfirewall   = no
        auto            = route
        right           = 88.88.88.88
        rightsubnet     = 10.20.10.0/24

Nov  2 00:01:00 ip-10-221-119-237 charon: 13[NET] received packet: from 176.12.106.30[4500] to 10.221.119.237[4500] (256 bytes)
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[ENC] parsed ID_PROT request 0 [ SA V V V V V V V V V ]
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received strongSwan vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received Cisco Unity vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received XAuth vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received DPD vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received NAT-T (RFC 3947) vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received draft-ietf-ipsec-nat-t-ike-03 vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received draft-ietf-ipsec-nat-t-ike-02 vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] received draft-ietf-ipsec-nat-t-ike-00 vendor ID
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[IKE] 88.88.88.88 is initiating a Main Mode IKE_SA
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[ENC] generating ID_PROT response 0 [ SA V V V ]
Nov  2 00:01:00 ip-10-221-119-237 charon: 13[NET] sending packet: from 10.221.119.237[4500] to 88.88.88.88[4500] (136 bytes)
Nov  2 00:01:00 ip-10-221-119-237 charon: 16[NET] received packet: from 88.88.88.88[4500] to 10.221.119.237[4500] (380 bytes)
Nov  2 00:01:00 ip-10-221-119-237 charon: 16[ENC] parsed ID_PROT request 0 [ KE No NAT-D NAT-D ]
Nov  2 00:01:00 ip-10-221-119-237 charon: 16[IKE] local host is behind NAT, sending keep alives
Nov  2 00:01:00 ip-10-221-119-237 charon: 16[ENC] generating ID_PROT response 0 [ KE No NAT-D NAT-D ]
Nov  2 00:01:00 ip-10-221-119-237 charon: 16[NET] sending packet: from 10.221.119.237[4500] to 88.88.88.88[4500] (396 bytes)
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[NET] received packet: from 88.88.88.88[4500] to 10.221.119.237[4500] (76 bytes)
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[ENC] parsed ID_PROT request 0 [ ID HASH ]
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[CFG] looking for pre-shared key peer configs matching 10.221.119.237...88.88.88.88[88.88.88.88]
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[CFG] selected peer config "img-10-10-0-0-16-10-221-112-0-22"
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[IKE] IKE_SA img-10-10-0-0-16-10-221-112-0-22[361] established between 10.221.119.237[52.56.172.18]...88.88.88.88[88.88.88.88]
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[IKE] scheduling reauthentication in 27793s
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[IKE] maximum IKE_SA lifetime 28333s
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[ENC] generating ID_PROT response 0 [ ID HASH ]
Nov  2 00:01:00 ip-10-221-119-237 charon: 11[NET] sending packet: from 10.221.119.237[4500] to 88.88.88.88[4500] (92 bytes)
Nov  2 00:01:00 ip-10-221-119-237 charon: 08[NET] received packet: from 88.88.88.88[4500] to 10.221.119.237[4500] (444 bytes)
Nov  2 00:01:00 ip-10-221-119-237 charon: 08[ENC] parsed QUICK_MODE request 2720731132 [ HASH SA No KE ID ID ]
Nov  2 00:01:00 ip-10-221-119-237 charon: 08[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
Nov  2 00:01:00 ip-10-221-119-237 charon: 08[ENC] generating QUICK_MODE response 2720731132 [ HASH SA No KE ID ID ]
Nov  2 00:01:00 ip-10-221-119-237 charon: 08[NET] sending packet: from 10.221.119.237[4500] to 88.88.88.88[4500] (460 bytes)
Nov  2 00:01:01 ip-10-221-119-237 charon: 07[NET] received packet: from 88.88.88.88[4500] to 10.221.119.237[4500] (444 bytes)
Nov  2 00:01:01 ip-10-221-119-237 charon: 07[ENC] parsed QUICK_MODE request 1236930267 [ HASH SA No KE ID ID ]
Nov  2 00:01:01 ip-10-221-119-237 charon: 07[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
Nov  2 00:01:01 ip-10-221-119-237 charon: 07[ENC] generating QUICK_MODE response 1236930267 [ HASH SA No KE ID ID ]
Nov  2 00:01:01 ip-10-221-119-237 charon: 07[NET] sending packet: from 10.221.119.237[4500] to 88.88.88.88[4500] (460 bytes)
Nov  2 00:01:01 ip-10-221-119-237 charon: 15[NET] received packet: from 88.88.88.88[4500] to 10.221.119.237[4500] (444 bytes)
Nov  2 00:01:01 ip-10-221-119-237 charon: 15[ENC] parsed QUICK_MODE request 1502915672 [ HASH SA No KE ID ID ]
Nov  2 00:01:01 ip-10-221-119-237 charon: 15[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
Nov  2 00:01:01 ip-10-221-119-237 charon: 15[ENC] generating QUICK_MODE response 1502915672 [ HASH SA No KE ID ID ]
Nov  2 00:01:01 ip-10-221-119-237 charon: 15[NET] sending packet: from 10.221.119.237[4500] to 88.88.88.88[4500] (460 bytes)
Nov  2 00:01:01 ip-10-221-119-237 charon: 05[NET] received packet: from 88.88.88.88[4500] to 10.221.119.237[4500] (444 bytes)
Nov  2 00:01:01 ip-10-221-119-237 charon: 05[ENC] parsed QUICK_MODE request 3138312643 [ HASH SA No KE ID ID ]
Nov  2 00:01:01 ip-10-221-119-237 charon: 05[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
...skipping...
Nov  2 00:01:01 ip-10-221-119-237 charon: 09[ENC] invalid HASH_V1 payload length, decryption failed?
Nov  2 00:01:01 ip-10-221-119-237 charon: 09[ENC] could not decrypt payloads
Nov  2 00:01:01 ip-10-221-119-237 charon: 09[IKE] message parsing failed
Nov  2 00:01:01 ip-10-221-119-237 charon: 09[ENC] generating INFORMATIONAL_V1 request 1915578037 [ HASH N(PLD_MAL) ]
Nov  2 00:01:01 ip-10-221-119-237 charon: 09[NET] sending packet: from 10.221.119.237[4500] to 88.88.88.88[4500] (92 bytes)
Nov  2 00:01:01 ip-10-221-119-237 charon: 09[IKE] QUICK_MODE request with message ID 2720731132 processing failed
Nov  2 00:01:01 ip-10-221-119-237 charon: 12[NET] received packet: from 88.88.88.88[4500] to 10.221.119.237[4500] (76 bytes)
Nov  2 00:01:01 ip-10-221-119-237 charon: 12[ENC] invalid HASH_V1 payload length, decryption failed?
Nov  2 00:01:01 ip-10-221-119-237 charon: 12[ENC] could not decrypt payloads
Nov  2 00:01:01 ip-10-221-119-237 charon: 12[IKE] message parsing failed

**Comment by Thermi** [comment]
looks OK. Maybe you need to increase `charon.max_ikev1_exchanges` from its default of `3` to a higher number like `10` or so. Let me know if that works for you.

**Comment by doug-tucker-nav** [comment]
I would see that sequence 6 times before it would then go through some other things before finally sending a delete.  I modified the value to 10 as requested, though I'm skeptical this is the issue.  The customer is on a different timezone, so it will be tomorrow before they are able to try again.

**Comment by Devstellar** [comment]
`Nov 2 00:01:00 ip-10-221-119-237 charon: 13[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048`

Could be related to an issue I posted in the mailing list?
https://lists.strongswan.org/pipermail/users/2022-September/015440.html

You can test changing to SHA1

**Reply by doug-tucker-nav** [reply]
Thanks, but we are both using a PSK that seems to be working just fine.  If those don't match there will be no tunnel establishment, and phase1 works when either of us initiate, and phase 2 completes when I do.
